core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -63,7 +63,7 @@ public DataExpressionSource(final ValueExpression dataExpression, final int inde
 
     @Override
     public String toString() {
-        return dataExpression.toString() + "[" + index + "](" + graph + "," + encoding + ")";
+        return getClass().getSimpleName() + "(" + dataExpression.toString() + "[" + index + "](" + graph + "," + encoding + "))";
```

Comment:
With large string concatenations (> 5 * `+`) I tend to use String.formats, such that you see the overal idea of the string:
````java
String.format("%s(%s[%d](%s,%s))", getClass.getSimpleName(), dataExpression.toString(), index, graph, encoding);
````
But that is very personal, because I also understand that it is harder to see which `%s` matches with which string.